### DISCUSSION TITLE
Is possible distinguish between homographs from context in Polish?

### DISCUSSION THREAD

**Discussion by andr1972** [explosion/spaCy#12388] [Help: Other Questions] [answered]
Word "bez" in Polish has meanings:
- without
- lilac (especially 'czarny bez' = 'black lilac')

If I analyze
```
txt = "Ciocia kupiła czarny bez"
print("Base word:", [token.lemma_ for token in doc])
print("pos:", [token.pos_ for token in doc])
```
we have:
Base word: ['ciocia', 'kupić', 'czarny', 'bez']
pos: ['NOUN', 'VERB', 'ADJ', 'ADP']

last -  'ADP'  should be NOUN, not ADP
In Polsh is possible context meanings?

**Comment by thomashacker** [comment, marked as answer]
Hey, thanks for your post!
Can you provide more information about which polish spaCy model you are using? 
You could try to use other pre-trained models and see if the results are improving.

Generally, ML models are mostly never 100% and do misclassification sometimes, depending on the data a model was trained on.
We have a thread about this [issue](https://github.com/explosion/spaCy/issues/3052), which goes into that topic with more detail.

**Comment by andr1972** [comment]
I try pl_core_news_sm,pl_core_news_md and pl_core_news_lg

**Reply by thomashacker** [reply]
The problem is that the datasets the polish models were trained on aren't as resourceful as other languages like English. This could explain why their performance is limited and why misclassifications occur more often. Feel free to add your example [to this master thread](https://github.com/explosion/spaCy/issues/3052).
